### ISSUE TITLE
How to append data to existing data in handsontable?

### ISSUE THREAD

**Issue by mukta88** [handsontable/handsontable#2343]
Hello all,

I had already added this issue under
https://github.com/handsontable/handsontable/issues/505
But that issue is closed so again adding it here as a new issue.

I have 1000 number of data in a handsontable. on window scroll, I need to append next 1000 records to current  handsontable.
I used, loaddata with new data, but it replaced first 1000 records with new 1000 records,

How can I do that?
Thanks in advance.

**Comment by budnix** [comment]
If you need to append more data to the current handsontable instance you can simply push next row into data source like in this demo http://jsfiddle.net/yk64yv8q/4/.

**Comment by mukta88** [comment]
Thanks you very much for the solution :+1:

**Comment by kameelyan** [comment]
This no longer works in the latest version. You can see your above fiddle for an example, also listed here: http://jsfiddle.net/yk64yv8q/4/

**Comment by randy-argonillo** [comment]
Hi,

I've experimented the sample code although it is correct that hot.getData() no longer works as expected but I found out that you can now use hot.getSourceData() instead in pushing additional data.

**Comment by abtx** [comment]
My `hot.getSourceData()` will return correct list of appended elements, but my table doesn't render it properly, i.e. it appends one batch of data and then it's creating empty space. Anyone had this issue?

**Comment by AMBudnik** [comment]
Hi guys, can you share a demo where this issue can be replicable?

**Comment by abtx** [comment]
Hi Alex, here's [a link](www.well-off-circle.surge.sh) with implementation I'm having trouble with.

**Comment by AMBudnik** [comment]
Thanks for sharing @abtx that's also the demo with the dropping headers. I remember it, but can you share a JSFiddle/JSBin example where I'll be able to put some of my code and (what's even more important) simplify the example.

**Comment by abtx** [comment]
Sure Alex, I'll prep a demo for you in the coming days. Thanks!

**Comment by abtx** [comment]
Ok Alex, I've recreated my table in [JSFiddle](https://jsfiddle.net/abtx/pcwudrmc/10718/) and it almost works fine for me. There is one other issue that happens now. Please try appending data 3-4 times, you'll notice that you can't scroll to the very bottom at some point. 

What I did to fix the previous issue is use [this CDN](https://docs.handsontable.com/2.0.0/bower_components/handsontable/dist/handsontable.full.css), which should work fine with bower but for some reason using this link explicitly makes it work 😳😂

**Comment by abtx** [comment]
I've created the [JSFIddle](https://jsfiddle.net/abtx/pcwudrmc/10718/), and the header and sizing issues are gone, but if you click the `load more data` button twice you won't be able to scroll to the bottom and see it again. 

What I did to make this work is - weirdly enough - remove my bower reference to HOT and replace with ones used in the JSFiddle. My `bower.json` specified version 2.0 or higher, but apparently this bower cdn has a different version where these two previous issues are eradicated. 

Could you please look at the obstructed button issue? Thanks Alex!!! 🙌

**Comment by abtx** [comment]
@AMBudnik Alex, I noticed that the button will be visible but only after some time, i.e. you can't scroll to the bottom of the page, leave some time and the page seems to allow me to scroll to the bottom. I also noticed misalignment in rows after loading more data. Cheers!
<img width="233" alt="screen shot 2018-04-17 at 15 33 18" src="https://user-images.githubusercontent.com/5509711/38855066-05108b24-4255-11e8-9448-63687a1fbe7f.png">

**Comment by AMBudnik** [comment]
Wow, so many updates on that issue :) 

I'll play with the example and share my thoughts. Firstly it is noticeable that the data is loading very slow. I even jumped to the console looking for some errors but table finally showed up.

**Comment by abtx** [comment]
@AMBudnik I can confirm this issue happens on Opera only (certainly not a blocker) 💯

**Comment by abtx** [comment]
Opera is just a Chrome, so should be consistent. It actually works now on Opera too. IE? Didn't even check (and don't want to) 😂

**Comment by AMBudnik** [comment]
It works now @abtx? What has changed?

**Comment by abtx** [comment]
As far as I remember it started working after I changed source from bower to a cdn (https://docs.handsontable.com/2.0.0/bower_components/handsontable/dist/handsontable.full.js)

**Comment by toanpv-2417** [comment]
@AMBudnik how you custom loadmore button inside handsontable ? can you give me some suggestion . Thanks !

**Comment by AMBudnik** [comment]
Hi @toanpv-2417 

Could you share the requirements details with me on the email at support@handsontable.com?
This issue is from 2015, we wouldn't want the author to get any unnecessary notifications, not connected to the issue report.
